src/api/middleware/auth.js

```diff
@@ -78,15 +78,15 @@ export const hasAdminPrivileges = async (req, res, next) => {
 };
 
 export const validToken = (req, res, next) => {
-    const decoded = verifyAndDecodeToken(req.params.token, config.jwt_secret);
-
-    if (!decoded) {
-        return next(new APIError(HTTPStatus.FORBIDDEN, ErrorTypes.FORBIDDEN, ValidationReasons.INVALID_TOKEN));
-    }
+    try {
+        const decoded = verifyAndDecodeToken(req.params.token, config.jwt_secret, next);
 
-    storeInLocals(req, {
-        token: decoded,
-    });
+        storeInLocals(req, {
+            token: decoded,
+        });
 
-    return next();
+        return next();
+    } catch (err) {
+        return next(err);
+    }
```

Comment:
Why wrapped in try catch?

Reply:
We were not aware that calling next with an error would skip all the handlers that can't handle such error. We are removing the try catch.